pkg/agent/flowexporter/exporter/exporter.go

```diff
@@ -298,19 +295,23 @@ func (exp *flowExporter) sendFlowRecords() error {
 				}
 				exp.numDataSetsSent = exp.numDataSetsSent + 1
 			}
-			// If the connection is not present in the conntrack table anymore,
-			// then we delete the connection in connection store and record store.
-			// Please note that our record cache policy is that we keep the record in
-			// the record map as long as the corresponding connection is present in
-			// the conntrack table.
-			if !record.Conn.IsPresent {
-				klog.V(2).Infof("Deleting the inactive connection with key: %v from both connection map and record map", key)
-				if err := exp.connStore.DeleteConnectionByKey(key); err != nil {
+
+			if flowexporter.IsConnectionDying(&record.Conn) {
+				// If the connection is in the dying state, then we will set the flow record
+				// to nil after sending it.
+				if err := exp.flowRecords.SetFlowRecordToNilWithoutLock(key); err != nil {
```

Comment:
I think this has to be modified as explained offline. This will not lead to the proper deletion of flow records and connections.